src/execution_plan/ops/op_foreach.c

```diff
@@ -67,7 +67,7 @@ static Record _handoff(OpForeach *op) {
 	ASSERT(op->records != NULL);
 
 	Record r = NULL;
-	if(array_len(op->records)) {
+	if(array_len(op->records) > 0) {
 		r = array_pop(op->records);
 	}
```

Comment:
we could get rid of this condition `if(array_len(op->records) > 0)` if we manually insert a `NULL` value to `op->records` when this operation is constructed.
when the NULL value is popped and return execution terminates

Reply:
That sounds great 👍